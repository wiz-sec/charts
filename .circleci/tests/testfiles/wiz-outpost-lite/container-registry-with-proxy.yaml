agent:
  secretName: outpost-lite-agent-creds
  clientId: "client-1"
  clientSecret: "secret-2"
  outpostId: "my-outpost-id"

runners:
  container-registry:
    enabled: true


httpProxyConfiguration:
  enabled: true
  httpProxy: my-http-proxy
  httpsProxy: my-https-proxy
  noProxy:
    - no-proxy-1
    - no-proxy-2
  caCertificate: |
    --- BEGIN CERTIFICATE ---
    The cert
    --- END CERTIFICATE ---