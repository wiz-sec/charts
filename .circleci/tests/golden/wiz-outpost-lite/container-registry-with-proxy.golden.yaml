---
# Source: wiz-outpost-lite/templates/credentials.secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: outpost-lite-agent-creds
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-outpost-lite
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1.0"
    app.kubernetes.io/managed-by: Helm
stringData:
  clientId: "client-1"
  clientSecret: "secret-2"
---
# Source: wiz-outpost-lite/templates/httpProxyConfiguration.secret.yaml
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: wiz-http-proxy-configuration
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-outpost-lite
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1.0"
    app.kubernetes.io/managed-by: Helm
  annotations:
    meta.helm.sh/release-name: "release-test"
    meta.helm.sh/release-namespace: "release-helm-namespace"
data:
  httpProxy: "bXktaHR0cC1wcm94eQ=="
  http-proxy: "bXktaHR0cC1wcm94eQ=="
  httpsProxy: "bXktaHR0cHMtcHJveHk="
  https-proxy: "bXktaHR0cHMtcHJveHk="
  no-proxy-address: "bm8tcHJveHktMSBuby1wcm94eS0y"
  no-proxy-address-cs: "bm8tcHJveHktMSxuby1wcm94eS0y"
  noProxyAddress: "bm8tcHJveHktMSxuby1wcm94eS0y"
  noProxyAddressSpaceSepareted: "bm8tcHJveHktMSBuby1wcm94eS0y"
  caCertificate: "LS0tIEJFR0lOIENFUlRJRklDQVRFIC0tLQpUaGUgY2VydAotLS0gRU5EIENFUlRJRklDQVRFIC0tLQo="
---
# Source: wiz-outpost-lite/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-test-wiz-outpost-lite-container-registry
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-outpost-lite
    app.kubernetes.io/instance: release-test
    wiz.io/runner: "container-registry"
    app.kubernetes.io/version: "0.1.0"
    app.kubernetes.io/managed-by: Helm
spec:
  type: ClusterIP
  ports:
    - port: 9090
      targetPort: metrics
      protocol: TCP
      name: metrics
  selector:
    app.kubernetes.io/name: wiz-outpost-lite
    app.kubernetes.io/instance: release-test
    wiz.io/runner: "container-registry"
---
# Source: wiz-outpost-lite/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-test-wiz-outpost-lite-container-registry
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-outpost-lite
    app.kubernetes.io/instance: release-test
    wiz.io/runner: "container-registry"
    app.kubernetes.io/version: "0.1.0"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: wiz-outpost-lite
      app.kubernetes.io/instance: release-test
      wiz.io/runner: "container-registry"
  template:
    metadata:
      annotations:
        container.apparmor.security.beta.kubernetes.io/wiz-outpost-lite-container-registry: unconfined
      labels:
        app.kubernetes.io/name: wiz-outpost-lite
        app.kubernetes.io/instance: release-test
        wiz.io/runner: "container-registry"
    spec:
      restartPolicy: Always
      containers:
        - name: wiz-outpost-lite-container-registry
          image: "wizio.azurecr.io/outpost-lite-runner-container-registry:0.1-latest"
          command: [ "/entrypoint"]
          imagePullPolicy: Always
          env:
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: K8S_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: K8S_POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: OUTPOST
            value: "1"
          - name: WIZ_OUTPOST_ID
            value: "my-outpost-id"
          - name: WIZ_OUTPOST_RUNNER_ID
            value: "container-registry"
          - name: WIZ_CLIENT_ID
            valueFrom:
              secretKeyRef:
                name: outpost-lite-agent-creds
                key: clientId
          - name: WIZ_CLIENT_SECRET
            valueFrom:
              secretKeyRef:
                name: outpost-lite-agent-creds
                key: clientSecret
          - name: OUTPOST_LITE_RUNNER_REGION
            value: "partition-1"
          - name: OUTPOST_LITE_RUNNER_METRICS_PORT
            value: "9090"
          - name: OUTPOST_LITE_RUNNER_AUTO_UPDATE
            value: "1"
          - name: http_proxy
            valueFrom:
              secretKeyRef:
                name: "wiz-http-proxy-configuration"
                key: "httpProxy"
                optional: true
          - name: HTTP_PROXY
            valueFrom:
              secretKeyRef:
                name: "wiz-http-proxy-configuration"
                key: "httpProxy"
                optional: true
          - name: https_proxy
            valueFrom:
              secretKeyRef:
                name: "wiz-http-proxy-configuration"
                key: "httpsProxy"
                optional: true
          - name: HTTPS_PROXY
            valueFrom:
              secretKeyRef:
                name: "wiz-http-proxy-configuration"
                key: "httpsProxy"
                optional: true
          - name: no_proxy
            valueFrom:
              secretKeyRef:
                name: "wiz-http-proxy-configuration"
                key: "noProxyAddress"
                optional: true
          - name: NO_PROXY
            valueFrom:
              secretKeyRef:
                name: "wiz-http-proxy-configuration"
                key: "noProxyAddress"
                optional: true
          - name: SSL_CERT_DIR
            value: "/usr/local/share/ca-certificates/:/certificates/"
          ports:
            - name: metrics
              containerPort: 9090
              protocol: TCP
          resources:
            limits:
              memory: 4396M
            requests:
              memory: 1024M
          securityContext:
            capabilities:
              add:
              - SYS_ADMIN
            seLinuxOptions:
              type: spc_t
          volumeMounts:
            - mountPath: /var/wiz
              name: working-dir
            - mountPath: /usr/local/share/ca-certificates/
              name: ca-certificate
              readOnly: true
      terminationGracePeriodSeconds: 30
      volumes:
        - name: working-dir
          emptyDir: {}
        - name: ca-certificate
          secret:
            defaultMode: 420
            secretName: "wiz-http-proxy-configuration"
            items:
              - key: caCertificate
                path: root.crt
            optional: true
